variables.tf

```diff
@@ -2,10 +2,181 @@
+    condition = anytrue([
+      var.mongodb_version == null,
+      var.mongodb_version == "4.4",
+      var.mongodb_version == "4.2",
+    ])
+    error_message = "Version must be 4.2 or 4.4. If null, the current default ICD mongodb version is used"
+  }
+}
+
+variable "resource_group_id" {
+  type        = string
+  description = "The ID of the IMB Cloud resource group where you want to create the instance"
+}
+
+variable "tags" {
+  type        = list(any)
+  description = "A list of tags that you want to add to your instance"
+  default     = []
+}
+
+variable "endpoints" {
+  type        = string
+  description = "Endpoints available to the mongodb instance (public, private, public-and-private)"
+  default     = "private"
+
+  validation {
+    condition     = can(regex("public|private|public-and-private", var.endpoints))
+    error_message = "Valid values are (public, private, or public-and-private)."
+  }
+}
+
+variable "key_protect_key_crn" {
+  type        = string
+  description = "(Optional) The root key CRN of a Key Management Service like Key Protect or Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption. If `null`, database is encrypted by using randomly generated keys. See https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-key-protect&interface=ui#key-byok for current list of supported regions for BYOK"
+  default     = null
+}
+
+variable "backup_encryption_key_crn" {
+  type        = string
+  description = "(Optional) The CRN of a key protect key, that you want to use for encrypting disk that holds deployment backups. If null, will use 'key_protect_key_crn' as encryption key. If 'key_protect_key_crn' is also null database is encrypted by using randomly generated keys."
+  default     = null
+}
+
+variable "memory_mb" {
+  type        = number
+  description = "Memory available to the mongodb instance"
+  default     = 1024
+
+  validation {
+    condition = alltrue([
+      var.memory_mb >= 1024,
+      var.memory_mb <= 114688
+    ])
+    error_message = "Member group memory must be >= 1024 and <= 114688 in increments of 128."
+  }
+}
+
+variable "disk_mb" {
+  type        = number
+  description = "Disk available to the mongodb instance"
+  default     = 20480
+
+  validation {
+    condition = alltrue([
+      var.disk_mb >= 5120,
+      var.disk_mb <= 4194304
+    ])
+    error_message = "Member group disk must be >= 5120 and <= 4194304 in increments of 1024."
+  }
+}
+
+variable "cpu_count" {
+  type        = number
+  description = "Number of CPU cores available to the mongodb instance"
+  default     = 7
+
+  validation {
+    condition = alltrue([
+      var.cpu_count >= 6,
+      var.cpu_count <= 28
+    ])
+    error_message = "Number of cpus must be >= 6 and <= 28 in increments of 1."
+  }
+}
+
+
+variable "members" {
+  type        = number
+  description = "Allocated number of members"
+  default     = 3
+
+  validation {
+    condition = alltrue([
+      var.members == 3
```

Comment:
I think this is a mistake? Would make it so that members could only ever be exactly "3"?